corgi/tasks/pnc.py

```diff
@@ -30,4 +36,82 @@ def slow_fetch_pnc_sbom(purl: str, product_data, build_data, sbom_data) -> None:
     response.raise_for_status()
     logger.info(f"SBOM fetched for purl {purl}")
 
-    parse_pnc_sbom(response.json())
+    # Make sure the SBOM is valid
+    sbom = SbomerSbom(response.json()["sbom"])
+
+    # Create a build for the root component
+    root_build, created = SoftwareBuild.objects.get_or_create(
+        build_id=sbom.components["root"]["meta_attr"]["pnc_build_id"],
+        build_type=SoftwareBuild.Type.PNC,
+    )
+
+    # Create ProductComponentRelation if it doesn't already exist
+    if created:
+        pcr, _ = ProductComponentRelation.objects.get_or_create(
+            build_id=root_build.build_id,
+            build_type=SoftwareBuild.Type.PNC,
+            software_build__isnull=root_build,
+            defaults={
+                "type": ProductComponentRelation.Type.SBOMER,
+            },
+        )
+
+    # Create components
+    components = {}
+    for bomref, component in sbom.components.items():
+        defaults = {"namespace": component["namespace"], "meta_attr": component["meta_attr"]}
+
+        if "related_url" in component:
+            defaults["related_url"] = component["related_url"]
+        if bomref == "root":
+            defaults["software_build"] = root_build
+
+        components[bomref], _ = Component.objects.update_or_create(
+            type=Component.Type.MAVEN,
```

Comment:
In the context of this collector, we'll only get manifests which document PNC builds. In the manifest for Quarkus 2.13.8, there are two upstream components (protoc and protoc-gen-rpc-java) that are executables and not Java artifacts. It might be worth writing some corgi-wide function to infer component type from the purl type parameter.

Reply:
Created https://issues.redhat.com/browse/CORGI-793 to follow up on this later.